src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -8,111 +8,69 @@
 using System.Security.Cryptography.X509Certificates;
 using Microsoft.Win32.SafeHandles;
 
+using SafeSslHandle = Interop.libssl.SafeSslHandle;
+
 internal static partial class Interop
 {
     internal static class OpenSsl
     {
-        #region structures
-        [StructLayout(LayoutKind.Sequential)]
-        private struct SslContext
-        {         
-            internal IntPtr sslPtr;
-            internal IntPtr readBioPtr;
-            internal IntPtr writeBioPtr;
-            internal bool isServer;
-        }
-        #endregion
-
         #region internal methods
 
-
         //TODO (Issue #3362) Set remote certificate options
-        internal static IntPtr AllocateSslContext(long options, SafeX509Handle certHandle, SafeEvpPKeyHandle certKeyHandle, bool isServer, bool remoteCertRequired)        
+        internal static SafeSslHandle AllocateSslContext(long options, SafeX509Handle certHandle, SafeEvpPKeyHandle certKeyHandle, bool isServer, bool remoteCertRequired)
         {
-            SslContext sslContext = new SslContext
-            {
-                isServer = isServer,
-            };
+            SafeSslHandle context = null;
 
             try
             {
                 IntPtr method = GetSslMethod(isServer, options);
 
-                IntPtr contextPtr = libssl.SSL_CTX_new(method);
-
-                if (IntPtr.Zero == contextPtr)
-                {
-                    throw CreateSslException("Failed to allocate SSL/TLS context");
-                }
-
-                libssl.SSL_CTX_ctrl(contextPtr, libssl.SSL_CTRL_OPTIONS, options, IntPtr.Zero);
-
-                libssl.SSL_CTX_set_quiet_shutdown(contextPtr, 1);
-
-                if (certHandle != null && certKeyHandle != null)
+                using (libssl.SafeSslContextHandle sslContext = new libssl.SafeSslContextHandle(method))
```

Comment:
Personally I find it a bit hard to read with `context` and `sslContext` and they're different types.  Perhaps `context` could be `ssl`, or `sslHandle`, or something within this method?